### PR TITLE
chore(core): replace admin-only checks with permissions

### PR DESCRIPTION
**Pull request by glasstiger**
replacing admin only check with permission check hooks.

required by https://github.com/questdb/questdb-enterprise/pull/521

### CODE REVIEW

**Comment by glasstiger** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 19 / 21 (90.48%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/rnd/RndLogFunctionFactory.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/SimulateCrashFunctionFactory.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/test/TestNPEFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/SimulateWarningsFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/table/HydrateTableMetadataFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/activity/QueryActivityFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/security/DenyAllSecurityContext.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/FlushQueryCacheFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/security/ReadOnlySecurityContext.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/DumpThreadStacksFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/DumpMemoryUsageFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/QueryRegistry.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|3|3|100.00%|

File: core/src/main/java/io/questdb/cairo/SecurityContext.java [hunk 1]
```diff
@@ -83,12 +78,16 @@ default void authorizeCancelQuery() {
 
     void authorizePGWire();
 
+    void authorizeQueryAdmin();
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/cairo/SecurityContext.java:81]: can we call this "SQL Engine Admin" ?

File: core/src/main/java/io/questdb/cairo/SecurityContext.java [hunk 2]
```diff
@@ -124,6 +123,8 @@ default void clear() {
         // no-op
     }
 
+    void failIfReadOnly();
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/cairo/SecurityContext.java:126]: lets remove this

File: core/src/test/java/io/questdb/test/griffin/engine/functions/activity/QueryActivityFunctionFactoryTest.java [hunk 3]
```diff
@@ -253,8 +243,8 @@ public String getPrincipal() {
 
     private static class UserContext extends ReadOnlySecurityContext {
         @Override
-        public void authorizeAdminAction() {
-            throw new CairoException();
+        public void authorizeQueryAdmin() {
+            throw CairoException.authorization().put("Access denied for ").put(getPrincipal()).put(" [QUERY ADMIN]");
```
Comment (bluestreak01) [inline core/src/test/java/io/questdb/test/griffin/engine/functions/activity/QueryActivityFunctionFactoryTest.java:247]: lets rename QUERY ADMIN to "SQL Engine Admin"

Comment (glasstiger) [reply core/src/test/java/io/questdb/test/griffin/engine/functions/activity/QueryActivityFunctionFactoryTest.java]: renamed it

File: core/src/main/java/io/questdb/cairo/SecurityContext.java (hunk 2 above)
Comment (glasstiger) [reply core/src/main/java/io/questdb/cairo/SecurityContext.java]: turned it into:
```
boolean isQueryCancellationAllowed();
```

this is much more descriptive now.

**Review by bluestreak01 — APPROVED** [review]

**Comment by glasstiger** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 19 / 21 (90.48%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/rnd/RndLogFunctionFactory.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/SimulateCrashFunctionFactory.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/test/TestNPEFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/SimulateWarningsFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/SecurityContext.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/table/HydrateTableMetadataFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/activity/QueryActivityFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/FlushQueryCacheFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/security/ReadOnlySecurityContext.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/DumpThreadStacksFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/DumpMemoryUsageFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/QueryRegistry.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|2|2|100.00%|

**Pull request merged**